File: Challenge 4.py
'''
Write a program that will work out the
distance travelled if the user enters in the
speed and the time.

Extension
Get the program to tell you the speed you would have to travel
at in order to go a distance within a certain time entered by the
user. 
'''

import keyboard

print('If you want me to tell you how far you traveled, press 1.')
print('If you want to find out how fast you need to drive to get somewhere on time, press 2. ')

def on_1():
    try:
        speed_1 = int(input('What was your speed an hour? '))
        time_1 = int(input('For how long have you been driving? Please input your time in minutes. '))
        distance_1 = (speed_1 / 60) * time_1
        print('Distance you have traveled is ' + str(round(distance_1, 2)) + ' km.')
    except ValueError:
        print('You need to use numbers.')
    
def on_2():
    try:
        distance_2 = int(input('How far do you want travel? Enter in km. '))
        time_2 = int(input('How long do you have to get there? Enter in minutes. '))
        speed_2 = distance_2/ time_2 * 60
        print('You need to drive ' + str(round(speed_2, 2)) + ' km/h to get there on time.')
    except ValueError:
        print('You need to use numbers.')

keyboard.add_hotkey('1', on_1)
keyboard.add_hotkey('2', on_2)

while True:
    try:
        if keyboard.wait('1'):
            break
        elif keyboard.wait('2'):
            break
    finally:
        break
        

File: Challenge 4.java
import java.util.InputMismatchException;
import java.util.Scanner;

class TravelAssistant {
    public static int getNumber () {
        int input = 0;
        boolean correct = true;
        while (correct) {
            try {
                System.out.print(">> ");
                Scanner sc_input = new Scanner(System.in);
                input = sc_input.nextInt();
                correct = false;
            } catch (InputMismatchException Integer) {
                System.out.println("You need to use numbers.");
            }
        }
        return input;
    }

    public static void main(String[] args) {
        System.out.println("How far you traveled, press 1.");
        System.out.println("How fast you need to drive to arrive on time, press 2.");
        System.out.println("Press 3 to Exit.");
        boolean run = true;
        while (run == true) {
            int choice = getNumber();
            if (choice == 1) {
                System.out.println("What was your speed?");
                double speed = getNumber();
                System.out.println("How long have you been driving for? In minutes.");
                double time = getNumber();
                double distance = (speed / 60) * time;
                System.out.println("Distance you have traveled is " + Math.round(distance) + " km.");
            }
            else if (choice == 2) {
                System.out.println("How far do you want to travel? in km.");
                double distance = getNumber();
                System.out.println("How long do you have to get there? In minutes.");
                double time = getNumber();
                double speed = (distance / time) * 60;
                System.out.println("You need to drive " + Math.round(speed) + " km/h to get there on time.");
            }
            else if (choice == 3) {
                run = false;
            }
            else {
                System.out.println("You need to choose 1, 2 or 3.");
            }
        }
    }
}
